pkg/sql/pgwire/server.go

```diff
@@ -82,7 +82,11 @@ var logSessionAuth = settings.RegisterBoolSetting(
 	"if set, log SQL session login/disconnection events (note: may hinder performance on loaded nodes)",
 	false).WithPublic()
 
-var maxNumConnections = settings.RegisterIntSetting(
+// TODO(alyshan): This setting is enforcing max number of connections with superusers not being affected by
```

Comment:
good call on the TODO. we should do this one:
> update the description to say "the maximum number of connections per gateway ... Superusers are not affected by this limit"

(our team can do this in a later PR if that's easier)

Reply:
In spirit of breather week:

https://github.com/cockroachdb/cockroach/pull/101177